rct229/ruleset_functions/baseline_systems/is_baseline_system_7.py

```diff
@@ -0,0 +1,154 @@
+    is_hvac_sys_preheat_fluid_loop_attached_to_boiler,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_preheat_fluid_loop_purchased_heating import (
+    is_hvac_sys_preheat_fluid_loop_purchased_heating,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_preheating_type_fluid_loop import (
+    is_hvac_sys_preheating_type_fluid_loop,
+)
+from rct229.utils.jsonpath_utils import find_exactly_one_with_field_value, find_one
+
+HEATING_SYSTEM = schema_enums["HeatingSystemOptions"]
+NOT_SYS_7 = "Not_Sys_7"
+SYS_7 = "Sys-7"
+SYS_7A = "Sys-7a"
+SYS_7B = "Sys-7b"
+SYS_7C = "Sys-7c"
+
+
+def is_baseline_system_7(rmi_b, hvac_b_id, terminal_unit_id_list, zone_id_list):
+    """
+    Get either Sys-7, Sys-7a, Sys-7b, Sys-7c or Not_Sys_7 string output which indicates whether the HVAC system is
+    ASHRAE 90.1 2019 Appendix G system 7 (VAV with Reheat), system 7a (system 7 with purchased CHW), system 7b (
+    system 7 with purchased heating), pr system 7c (system 7 with purchased heating and purchased CHW).
+
+    Parameters
+    ----------
+    rmi_b: JSON, To evaluate if the hvac system is modeled as either Sys-7, Sys-7a, Sys-7b, Sys-7c or Not_Sys_7 in the B_RMR.
+    hvac_b_id: String, The id of the hvac system to evaluate.
+    terminal_unit_id_list: List<String>, list of terminal unit IDs associated with the HVAC system to be evaluated. These are
+                            sent to this function from the master get_baseline_system_types function.
+    zone_id_list: List<String>, list of zone IDs associated with the HVAC system to be evaluated. These are sent to this
+                        function from the master get_baseline_system_types function.
+
+    Returns: The function returns either Sys-7, Sys-7a, Sys-7b, Sys-7c or Not_Sys_7 string output which indicates
+    whether the HVAC system is ASHRAE 90.1 2019 Appendix G system 7 (VAV with Reheat), system 7a (system 7 with
+    purchased CHW), system 7b (system 7 with purchased heating), pr system 7c (system 7 with purchased heating and
+    purchased CHW). -------
+    """
+    is_baseline_system_7_str = NOT_SYS_7
+
+    # Get the hvac system
+    hvac_b = find_exactly_one_with_field_value(
+        "buildings[*].building_segments[*].heating_ventilation_air_conditioning_systems",
+        "id",
+        hvac_b_id,
+        rmi_b,
+    )
+
+    # check if the hvac system has the required sub systems for system type 7
+    has_required_sys = (
+        hvac_b.get("preheat_system") is not None
+        and hvac_b.get("cooling_system") is not None
+        and
+        # heating system shall be None or heating system type shall be None (the heating_system_type =
+        # HEATING_SYSTEM.NONE)
+        (
+            hvac_b.get("heating_system") is None
+            or find_one(
+                f"heating_system[?(@.heating_system_type={HEATING_SYSTEM.NONE})]",
```

Comment:
With jsonpath2, the curly braces need double quotes around them:
```
f'heating_system[?(@.heating_system_type="{HEATING_SYSTEM.NONE}")]'
```
and that requires the use of single quotes for the outer string.